pkgs/development/python-modules/pika/default.nix

```diff
@@ -16,7 +17,7 @@ buildPythonPackage rec {
     sha256 = "f023d6ac581086b124190cb3dc81dd581a149d216fa4540ac34f9be1e3970b89";
   };
 
-  checkInputs = [ nose mock twisted tornado ];
+  checkInputs = [ gevent nose mock twisted tornado ];
```

Comment:
```suggestion
  propagatedBuildInputs = [ gevent tornado twisted ];

  checkInputs = [ nose mock ];
  
  checkPhase = ''
    PIKA_TEST_TLS=true nosetests
  '';
```

Reply:
The pypi package contains no tests.

Reply:
We should use `fetchFromGitHub` then.

Reply:
Thanks! I enabled tests now, but 1 test fails because of a ConnectionRefusedError.
The problem is, I cannot find any information in the log about which exact test is failing.
The stack trace starts with a wrapper class which doesn't look like an actual test.
I could use some advice here.

Reply:
That's because networking is not allowed in the sandbox. We should to disable this test and a comment with the reason.

Reply:
Thanks. That is what I was trying to do. But I couldn't find out which test is failing. Any advice? (see last comment)

Reply:
I patched `setup.cfg` to remove `stop = 1` (which makes nose stop at the first failure) to discover which test it is. Turns out it's most (all?) of them in `tests/acceptance`. Thus let's patch `setup.cfg` in the following way:
```nix
  postPatch = ''
    # don't stop at first test failure
    # don't run acceptance tests because they access the network
    # don't report test coverage
    substituteInPlace setup.cfg \
      --replace "stop = 1" "stop = 0" \
      --replace "tests=tests/unit,tests/acceptance" "tests=tests/unit"
      --replace "with-coverage = 1" "with-coverage = 0" \
  '';
```

And finally please do
```nix
  checkPhase = ''
    runHook preCheck

    PIKA_TEST_TLS=true nosetests

    runHook postCheck
  '';
```